src/interactive/Toaster.jsx

```diff
@@ -0,0 +1,150 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import CSSTransition from 'react-transition-group/CSSTransition';
+import TransitionGroup from 'react-transition-group/TransitionGroup';
+import cx from 'classnames';
+import Toast from './Toast';
+
+export const DELAY_TIME = 3000;
+const MARGINAL_DELAY = 1000; // each additional toast will wait this much longer than the previous one
+
+/**
+ * @module Toaster
+ */
+class Toaster extends React.PureComponent {
+	constructor(props) {
+		super(props);
+
+		this.timeouts = [];
+
+		this.cloneToast = this.cloneToast.bind(this);
+		this.dismissToast = this.dismissToast.bind(this);
+		this.clearTimeouts = this.clearTimeouts.bind(this);
+		this.restartTimeouts = this.restartTimeouts.bind(this);
+		this.setTimer = this.setTimer.bind(this);
+
+		this.state = {
+			toasts: this.props.toasts.map(this.cloneToast)
+		};
+	}
+
+	/**
+	 * @param {Object} dismissedToast - `Toast` component to remove from state
+	 * @returns undefined
+	 */
+	dismissToast(dismissedToast) {
+		this.setState({
+			toasts: this.state.toasts.filter(toast => dismissedToast.props.id !== toast.props.id)
+		});
+	}
+
+	/**
+	 * @param {Object} toast - `Toast` component to set a timeout on before passing to `dismissToast`
+	 * @returns undefined
+	 *
+	 * sets timers to delay a toast's dismissal
+	 */
+	setTimer(toast) {
+		this.timeouts.push(setTimeout(() => {
+			this.dismissToast(toast);
+		}, DELAY_TIME + (MARGINAL_DELAY * toast.key)));
+	}
+
+	/**
+	 * @returns undefined
+	 *
+	 * removes timers in order to prevent toasts from being automatically dismissed
+	 */
+	restartTimeouts() {
+		const toasts = this.state.toasts;
+
+		toasts.forEach((toast, i) => {
+			this.setTimer(toast);
+		});
+	}
+
+	/**
+	 * @returns undefined
+	 *
+	 * removes timers in order to prevent toasts from being automatically dismissed
+	 */
+	clearTimeouts() {
+		this.timeouts.forEach(clearTimeout);
+	}
+
+	componentWillUnmount() {
+		this.clearTimeouts();
+	}
+
+	componentDidMount() {
+		this.state.toasts && this.state.toasts.filter(t => t.props.autodismiss).map(this.setTimer);
+	}
+
+	componentWillReceiveProps(nextProps) {
+		const currentToasts = this.state.toasts;
+		const allToasts = currentToasts.concat(nextProps.toasts);
+
+		console.log('--------------------componentWillReceiveProps fired--------------------');
+
+		if (nextProps.toasts !== this.state.toasts) {
+			this.setState(() => ({ toasts: allToasts.map(this.cloneToast) }), () => {
+				this.state.toasts && this.state.toasts.filter(t => t.props.autodismiss).map(this.setTimer);
+			});
+		}
+	}
+
+	/**
+	 * @param {Object} Toast - `Toast` components to clone
+	 * @param {number} i - index of the `toast`
+	 * @returns {Array} `Toast` components with props from `Toaster`
+	 */
+	cloneToast(toast, i) {
+		const toastProps = {
+			key: i,
+			id: toast.props.id || `toast-${i}`,
```

Comment:
just realized that this `id` scheme won't work - since `cloneToast` is called in the constructor, the first toast will get `#toast-0`. If a new toast is added in `componentWillReceiveProps`, it will _also_ get `#toast-0`.

I think you can/should get rid of any references to the toast `id`. It looks like it's only used for `this.dismissToast`, and I think you can just use an object equality check rather than an ID check for that, since `dismissToast` takes a `toast` that should be a reference to the same thing that is in `state`

```js
dismissToast(dismissedToast) {
  toasts: this.state.toasts.filter(toast => dismissedToast !== toast) })
}
```